### PR TITLE
Change stake modal to text input

### PR DESCRIPTION
**Pull request by ianrowan**
## Link to Issue
Closes: #6766 

## Description of Changes
https://www.loom.com/share/61e1cc9e926f44228becc4195b310bb2
- Converts stake amound from text to input
- updates style for text input
- adds dynamic character size for input

-bonus: includes an optimization for suggested gas in txs + add base as a network fix 

## Test Plan
- Click buy stake, input amount via input
- test + and -
- attempt again on sell flow

## Other Considerations

- supports max characters of 8, this is far beyond any likely affordable eth amount

### CODE REVIEW

**Review by mhagel — APPROVED** [review]
tested. looks good

**Review by masvelio — CHANGES_REQUESTED** [review]
1. I bought one stake and I am getting 4e-8 ETH. Not sure if this issue is because of this PR or not, but it worked fine earlier.

VOTE WEIGHT

2. I am concerned about this idea in general as it has a lot of potential issues that are already visible, both on technical and visual sides. I would say it needs better thought and maybe input from design team. I am listing some pitfalls I run into
- I am able to delete value from the input and a. it becomes without value on blur and b. it gives me 2 console errors because fundamental rules of react inputs are violated

https://github.com/hicommonwealth/commonwealth/assets/14819225/80397a4a-3d3e-42f1-ad66-caed93dab27f

- I can type non numeric values 
image

- I can type very long numbers which breaks both design and functionality
image
image
image

- it looks bad on mobile in such cases (layout was not prepared for such a big numbers)
image
image


In my opinion, it cannot get in with this shape. I think we should
- use number input
- add proper validation for max number values
- figure out better UI for mobile or/and big numbers

File: packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx [hunk 1]
```diff
@@ -149,9 +150,17 @@ const StakeExchangeForm = ({
...
+    const inputValue = e.target.value;
+    const numericValue = inputValue.replace(/[^0-9]/g, '');
+    onSetNumberOfStakeToExchange(
+      !numericValue || inputValue === '' ? null : parseInt(numericValue),
+    );
+  };
+
+  const insufficientFunds = isBuyMode
+    ? parseFloat(userEthBalance) < parseFloat(buyPriceData?.totalPrice)
+    : numberOfStakeToExchange > stakeBalance;
```
[the remark this answers is not in the document]
Comment (ianrowan) [reply packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx:166]: given that users can potentially input sell amounts larger than their holdings the user is show this message in sell now

File: packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx [hunk 2]
```diff
@@ -36,6 +36,7 @@ import {
   CustomAddressOptionElement,
 } from './CustomAddressOption';
 
+import { CWTextInput } from 'client/scripts/views/components/component_kit/cw_text_input';
```
[the remark this answers is not in the document]
Comment (ianrowan) [reply packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx]: addressed

File: packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx [hunk 3]
```diff
@@ -191,6 +200,13 @@ const StakeExchangeForm = ({
     ? false
     : numberOfStakeToExchange >= stakeBalance;
 
+  const inputClassExpanded =
+    numberOfStakeToExchange?.toString().length > 6
+      ? 'number expand2'
+      : numberOfStakeToExchange?.toString().length > 3
+      ? 'number expand1'
+      : 'number';
```
[the remark this answers is not in the document]
Comment (ianrowan) [reply packages/commonwealth/client/scripts/views/modals/ManageCommunityStakeModal/StakeExchangeForm/StakeExchangeForm.tsx:211]: addressed

**Review by masvelio — APPROVED** [review]
added some css tweaks in latest commit

**Pull request merged**